### PR TITLE
gobject-introspection: Fix macos shared lib paths

### LINKED ISSUE #40599: ATK gobject introspection broken on macOS
## Issue description

Try running redshift-gtk & observe error.

The ATK .gir location is incorrect. There is an @rpath in the path. It looks like it’s a meson issue.

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

### PR DESCRIPTION
**Pull request by hamishmack**
Fixes #40599.  This is the similar to #46310, but updated for the gnome-3.30 branch.

I used `cat $(nix-build ./default.nix -A atk.dev --system x86_64-darwin)/share/gir-1.0/Atk-1.0.gir | grep shared-library` to confirm the issue was still present in the gnome-3.30 branch and to confirm that this patch fixes it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by LnL7** [comment]
@GrahamcOfBorg build gobject-introspection

**Comment by hamishmack** [comment]
I have rebased this on latest gnome-3.30 branch.

**Comment by samueldr** [comment]
There is, again, a need for a rebase.

@jtojnar this branch is pretty much yours; any opinions on the fix?

File: pkgs/development/libraries/gobject-introspection/default.nix [hunk 1]
```diff
@@ -40,6 +40,7 @@ stdenv.mkDerivation rec {
   setupHook = ./setup-hook.sh;
 
   patches = [
+    ./macos-shared-library.patch
```
Comment (jtojnar) [inline pkgs/development/libraries/gobject-introspection/default.nix]: Could you name the patch something like `macos-absolutize-relocatable-install-name` and add a coment with a brief description here?

**Comment by hamishmack** [comment]
Updated as suggested and rebased

**Review by matthewbauer — APPROVED** [review]

**Comment by jtojnar** [comment]
https://github.com/NixOS/nixpkgs/pull/46310 was merged already.

**Pull request closed without merging**